### PR TITLE
Remove unused methods from the old Kube client

### PR DESCRIPTION
**Pull request by stevekuznetsov**
Signed-off-by: Steve Kuznetsov <skuznets@redhat.com>

/assign @fejta 
/cc @cjwagner

### CODE REVIEW

**Comment by stevekuznetsov** [comment]
@fejta ping

File: prow/cmd/mkbuild-cluster/main.go [hunk 1]
```diff
@@ -238,7 +238,7 @@ func do(o options) error {
 		if err != nil {
 			return err
 		}
-		if _, err = c.ListPods("k8s-app=kube-dns"); err != nil {
+		if _, err = c.ListProwJobs("k8s-app=kube-dns"); err != nil {
```
Comment (krzyzacy) [inline prow/cmd/mkbuild-cluster/main.go]: (I don't think this will work?)

Comment (stevekuznetsov) [reply prow/cmd/mkbuild-cluster/main.go]: If build clusters don't have ProwJobs registered?

Comment (stevekuznetsov) [reply prow/cmd/mkbuild-cluster/main.go]: Yeesh

Comment (krzyzacy) [reply prow/cmd/mkbuild-cluster/main.go]: oh.. I guess this meant to check if there is a kubedns pod exist?

File: prow/cmd/mkbuild-cluster/main.go [hunk 2]
```diff
@@ -238,7 +238,7 @@ func do(o options) error {
...
 			return fmt.Errorf("authenticated client could not list pods: %v", err)
```
Comment (krzyzacy) [inline prow/cmd/mkbuild-cluster/main.go]: s/pods/prowjobs

File: prow/cmd/mkbuild-cluster/main.go (hunk 1 above)
Comment (stevekuznetsov) [reply prow/cmd/mkbuild-cluster/main.go]: Very tempted to remove this check... @fejta ?

Comment (krzyzacy) [reply prow/cmd/mkbuild-cluster/main.go]: actually - just use a pod client here instead?

File: prow/cmd/mkbuild-cluster/main.go [hunk 3]
```diff
@@ -234,12 +238,37 @@ func do(o options) error {
...
+		contextCfg, err := clientcmd.NewNonInteractiveClientConfig(*cfg, o.alias, &clientcmd.ConfigOverrides{}, nil).ClientConfig()
+		if err != nil {
+			return fmt.Errorf("create %s client: %v", o.alias, err)
+		}
+		client, err := kubernetes.NewForConfig(contextCfg)
 		if err != nil {
-			return err
+			return fmt.Errorf("could not create kubernetes client: %v", err)
 		}
-		if _, err = c.ListPods("k8s-app=kube-dns"); err != nil {
```
Comment (fejta) [inline prow/cmd/mkbuild-cluster/main.go]: The point of this check is to ensure the user has a ClusterRoleBinding for manipulating pods

File: prow/cmd/mkbuild-cluster/main.go [hunk 4]
```diff
@@ -234,12 +238,37 @@ func do(o options) error {
...
-			return fmt.Errorf("authenticated client could not list pods: %v", err)
+		if _, err = client.CoreV1().Pods("default").List(metav1.ListOptions{}); err != nil {
```
Comment (fejta) [inline prow/cmd/mkbuild-cluster/main.go]: Do we know there's a default namespace? On the other hand, We can't be 100% confident that kube-dns will exist.

**Pull request merged**